File: PetBag/src/model/Dog.ts
import { Owner } from "./Owner";
import { Pet } from "./Pet";

export class Dog extends Pet {
    
    id: string | undefined;
    
    constructor(owner: Owner, firstName: String, breed: String, age: number, isDog: boolean, weight: number, numOfDays: number, isGroomed: boolean){
        super(owner, firstName, breed, age, isDog, weight, numOfDays, isGroomed);
        
    }
    public setId(id: string): void{
        this.id = id;
    }
    
    
}

File: PetBag/src/main/java/com/codingwithkaylah/model/Dog.java
package com.codingwithkaylah.model;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Document(collection="dogs")
public class Dog extends Pet {

	@Id
	private String id;
	
	
	
	public Dog(String firstName, String breed, int age, int numOfDays, Owner owner, 
			boolean isDog, boolean isGroomed, double weight) {
		super(owner, firstName, breed, age, numOfDays, isDog, weight,
				isGroomed);
		
	}
	public Dog(String id, String firstName, String breed, int age, int numOfDays, Owner owner, 
			boolean isDog, boolean isGroomed, double weight) {
		super(owner, firstName, breed, age, numOfDays, isDog, weight,
				isGroomed);
		this.id = id;
		
	}
	public Dog() {}
	
	public String getId() {
		return id;
	}
	public void setId(String id) {
		this.id = id;
	};

}
